back/src/Guide/Database/Queries/Update.hs

```diff
@@ -181,6 +181,11 @@ updateItemRow itemId f = do
     let statement :: Statement (Uid Item, Bool) ()
         statement = [execute|UPDATE items SET deleted = $2 WHERE uid = $1|]
     lift $ HT.statement (itemId, new_itemRowDeleted) statement
+    if new_itemRowDeleted
```

Comment:
I have to say that I don't like how these functions are becoming too smart. It's not `updateItemRow`'s job to maintain database invariants. Something like `_categoryRowItemsOrder %~ (++ [itemId])` belongs in the business logic, not here – for instance, we would definitely want to insert the item ID somewhere else, not to the end of the category. (When restoring a deleted item we record its position and then restore back to that position.)

I think database queries should in general be pretty dumb and all logic should belong in the functions using those queries.

Reply:
Next commit refactor all but did not this one above.